frontend/src/metabase/collections/components/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback } from "react";
+
+import { ANALYTICS_CONTEXT } from "metabase/collections/constants";
+import { Item, Collection } from "metabase/collections/utils";
+
+import { EntityItemMenu } from "./ActionMenu.styled";
+
+type Props = {
+  className?: string;
+  item: Item;
+  collection: Collection;
+  onCopy: (items: Item[]) => void;
+  onMove: (items: Item[]) => void;
+};
+
+function ActionMenu({ className, item, collection, onCopy, onMove }: Props) {
+  const handlePin = useCallback(() => {
+    item.setPinned(false);
+  }, [item]);
+
+  const handleCopy = useCallback(() => {
+    onCopy([item]);
+  }, [item, onCopy]);
+
+  const handleMove = useCallback(() => {
+    onMove([item]);
+  }, [item, onMove]);
+
+  const handleArchive = useCallback(() => {
+    item.setArchived(true);
+  }, [item]);
+
+  return (
+    <div
+      className={className}
+      onClick={e => {
+        e.stopPropagation();
+        e.preventDefault();
```

Comment:
this is needed because this component is nested within a `<Link>`

Reply:
Is it worth adding a comment maybe?

Reply:
Added -- also, ended up replacing the `<div onClick={...}>` with `<EventSandbox>`. The `HoverMenu` isn't very accessible right now but at some point keyboard events ought to work on it and the `EventSandbox` will catch all bubbling events.